static/app/views/stories/useStoriesLoader.tsx

```diff
@@ -0,0 +1,61 @@
+import {useEffect, useState} from 'react';
+
+import storiesContext from 'sentry/views/stories/storiesContext';
+import type {ResolvedStoryModule} from 'sentry/views/stories/types';
+
+interface Props {
+  filename: string;
+}
+
+interface EmptyState {
+  error: undefined;
+  filename: undefined | string;
+  resolved: undefined;
+}
+
+interface ResolvedState {
+  error: undefined;
+  filename: string;
+
+  resolved: ResolvedStoryModule;
+}
+
+interface ErrorState {
+  error: Error;
+  filename: undefined | string;
+  resolved: undefined;
+}
+
+type State = EmptyState | ResolvedState | ErrorState;
```

Comment:
Is there some way we can discriminate these?

Reply:
they are, Based on `error` and `resolved` being set/unset. It was really complaining before, but when i brought in the separate `interface` keywords things like index.tsx got happy.

Reply:
Might be nice to just make it explicit by having a `state` key that is `'resolved' | 'error' | 'empty'`

discriminating based on on undefined can be kind of annoying sometimes

but it's fine either way I think